usaspending_api/submissions/management/commands/load_dabs_submission_window_schedule.py

```diff
@@ -0,0 +1,112 @@
+    false as is_quarter
+from
+    submission_window_schedule
+where
+    period_start >= '2020-07-17'
+"""
+
+# SQL to create Quarter Period Schedules using broker table
+# Only use periods 3, 6, 9, and 12 from table
+# Submission Due Date comes from 'certification_deadline' column
+QUARTER_SCHEDULE_SQL = """
+select
+    make_date(year, period, 1) - interval '5 months' as period_start_date,
+    make_date(year, period, 1) - interval '2 months' - interval '1 day' as period_end_date,
+    period_start as submission_start_date,
+    certification_deadline as certification_due_date,
+    certification_deadline as submission_due_date,
+    certification_deadline + interval '1 day' as submission_reveal_date,
+    year as submission_fiscal_year,
+    (period + 2) / 3 as submission_fiscal_quarter,
+    period as submission_fiscal_month,
+    true as is_quarter
+from
+    submission_window_schedule
+where
+    period % 3 = 0
+"""
+
+# Manually creating some month period records because they are
+# not represented in the broker table
+EXTRA_MONTH_SCHEDULES = [
+    DABSSubmissionWindowSchedule(
+        period_start_date="2020-04-01 00:00:00",
+        period_end_date="2020-04-30 00:00:00",
+        submission_start_date="2020-07-17 00:00:00",
+        submission_due_date="2020-07-30 00:00:00",
+        certification_due_date="2020-08-14 00:00:00",
+        submission_reveal_date="2020-07-31 00:00:00",
+        submission_fiscal_year=2020,
+        submission_fiscal_quarter=3,
+        submission_fiscal_month=7,
+        is_quarter=False,
+    ),
+    DABSSubmissionWindowSchedule(
+        period_start_date="2020-05-01 00:00:00",
+        period_end_date="2020-05-31 00:00:00",
+        submission_start_date="2020-07-17 00:00:00",
+        submission_due_date="2020-07-30 00:00:00",
+        certification_due_date="2020-08-14 00:00:00",
+        submission_reveal_date="2020-07-31 00:00:00",
+        submission_fiscal_year=2020,
+        submission_fiscal_quarter=3,
+        submission_fiscal_month=8,
+        is_quarter=False,
+    ),
+]
+
+
+class Command(BaseCommand):
+    help = "Update DABS Submission Window Schedule table based on Broker"
+
+    @transaction.atomic()
+    def handle(self, *args, **options):
+        logger.info("Creating broker cursor")
+        broker_cursor = connections["data_broker"].cursor()
+
+        logger.info("Running MONTH_SCHEDULE_SQL")
+        broker_cursor.execute(MONTH_SCHEDULE_SQL)
+
+        logger.info("Getting month schedule values from cursor")
+        month_schedule_values = dictfetchall(broker_cursor)
+
+        logger.info("Running QUARTER_SCHEDULE_SQL")
+        broker_cursor.execute(QUARTER_SCHEDULE_SQL)
+
+        logger.info("Getting quarter schedule values from cursor")
+        quarter_schedule_values = dictfetchall(broker_cursor)
+
+        logger.info("Deleting existing DABS Submission Window Schedule")
+        DABSSubmissionWindowSchedule.objects.all().delete()
```

Comment:
Performing a full refresh like this, you will run out of integer ids in about 80 thousand years (ish).  If you're ok with this then :+1:.

Reply:
I looked into getting this sequence reset, but there doesn't seem to be a great way to do this with the Django ORM without running a query directly against a cursor. These IDs won't be used anywhere will they @kbard?

Reply:
I don't know what the future holds, but hopefully they'll learn very quickly that the ids aren't stable.  Another option would be to generate a deterministic id.  I did something like this when I was writing some garbage queries for something else, but maybe it'll be of use to you: `submission_fiscal_year * 1000 + submission_fiscal_month * 10 + is_quarter::int`.

Bit of a hack, but it's deterministic, stable, and keeps it a nice tiny integer.

Reply:
I think deterministic ids are a good idea. I wasn't sure if it would manually allow me to set them because I am using the sequence, but I tested it out, and it's good to go. Pushing the change now

Reply:
If you're gonna do that, just remove the sequence.  Change the id to an IntegerField.